### ISSUE TITLE
The objective passing to the predict function

### ISSUE THREAD

**Issue by andy920262** [cjlin1/simpleNN#6]
In `train.py`, the variable `loss` is passed to the predict function. But the actual objective is `loss_with_reg`. I consider here should be `loss_with_reg` in the predict function.

![註解 2020-04-06 105111](https://user-images.githubusercontent.com/14959701/78519558-5df1a880-77f6-11ea-9a48-d37b0848dab4.png)
![3](https://user-images.githubusercontent.com/14959701/78519563-60540280-77f6-11ea-9f9d-21281b5022dc.png)

**Comment by quanpr** [comment]
That's designed on purpose. In my humble opinion, we seldom do regularization in the test phase, right?

**Comment by andy920262** [comment]
I'm not really sure if regularization should be concerned in the test phase. I just notice this when I work on professor Lin's OPTDL course. I find keras.Model.fit will use the identical objective in the test phase.

**Comment by quanpr** [comment]
If you think about it carefully, having the regularization term or not will not affect the test accuracy. You may refer to [feedforward.m](https://github.com/cjlin1/simpleNN/blob/master/MATLAB/cnn/feedforward.m) for details. I would suggest you discuss with Prof. Lin or TAs for their opinion.

**Comment by andy920262** [comment]
Thank you for the answer. I think you're right, this can be designed by purpose. I'll close this issue.
